src/components/UserProfile/index.jsx

```diff
@@ -101,8 +102,25 @@ class UserProfile extends React.Component {
     };
 
     const getMode = (name) => {
-      if (name === this.props.currentlyEditingField) return 'editing';
-      if (!this.props[name] || !this.props[name].length) return 'empty';
+      // If the prop doesn't exist, that means it hasn't been set (for the current user's profile)
+      // or is being hidden from us (for other users' profiles)
+      const propExists = this.props[name] != null && this.props[name].length > 0;
+      // If this isn't the current user's profile, then there are only two options:
+      // static or nothing. We use 'null' as a return value because the consumers of
+      // getMode render nothing at all on a mode of null.
+      if (!isCurrentUserProfile) {
+        return propExists ? 'static' : null;
+      }
+      // Otherwise, if this is the current user's profile and it's name matches the
```

Comment:
You can probably leave out this comment and the two below it, since the code is pretty clear.  If you do keep them, small typo, it's -> its.

Reply:
I have this vague philosophy that as soon as a method gets too complex to figure out what's going on at a glance, it's time to comment the hell out of it. ;) 

But yes, fair.  That part is pretty self-explanatory.

For years I implicitly understood its vs. it's.  Somehow in the past month that appears to have gone out the window. :)

Reply:
Depends on what kind of complexity.  In general comments should explain *why* you're doing something.  If what you need to explain is *what* you're doing, I'd reach first for well named intermediary variables, well named methods, and other ways to communicate through the code itself.  Comments are sort of worse than tests, in that when they become out of date or wrong, they don't even break to let you know!

Reply:
That said, yeah, sometimes you're just stuck with a long list of things you need to do in order and comments can at least signpost the main chunks.  I don't think this method has hit that point though-- it's still young and spry.  ;-)